spec/support/model_helpers.rb

```diff
@@ -0,0 +1,35 @@
+# use like this:
+# it { is_expected.to have_validation :valid_date_presence,
+#                                     :on => :start_date,
+#                                     :options => [:allow_blank, :allow_nil]
+#                                     }
+RSpec::Matchers.define :have_validation do |validator, params|
+  validator_class = eval(validator.to_s.camelize + "Validator")
+
+  match do |model|
+    @_validators = model.class.validators.select { |v| (v.class == validator_class) && (v.attributes.include? params[:on]) }
+    validator_included_once? && (validator_receives_correct_options? params[:options])
```

Comment:
Thanks! And thanks for the reviews!

Yeah, I know, it is strange but from inside `validator_receives_correct_options?` method there is only access to instance variables. 
There is no access to `params`, `validator_class` and not even `model`.